lib/xctest.js

```diff
@@ -0,0 +1,185 @@
+import { startHouseArrestService, startTestmanagerdService, startInstallationProxyService, startInstrumentService } from './services';
+import { INSTRUMENT_CHANNEL } from './instrument';
+import { TESTMANAGERD_CHANNEL } from './testmanagerd';
+import { DTXMessageAuxBuffer } from './instrument/headers';
+import { NSUUID, XCTestConfiguration } from './instrument/transformer/nskeyed';
+import { uuidV4 } from '@appium/support/build/lib/util';
+import { getOSVersion } from './utilities';
+import log from './logger';
+import B from 'bluebird';
+
+class Xctest {
+    constructor(udid, xctestBundleId, targetBundleId = undefined, productModuleName = 'WebDriverAgentRunner') {
+        this.udid = udid;
+        this.running = false;
+        this._executing = false;
+        this.xctestBundleId = xctestBundleId;
+        this.targetBundleId = targetBundleId;
+        this.productModuleName = productModuleName;
+    }
+
+    async _launchAppRunner(majorVersion, sessionIdentifier) {
+        const { PROCESS_CONTROL } = INSTRUMENT_CHANNEL;
+        const installation = await startInstallationProxyService(this.udid);
+        let lookupResult;
+        try {
+            await installation.lookupApplications({ bundleIds: [this.xctestBundleId] });
+        } finally {
+            installation.close();
+        }
+        const appInfo = lookupResult[this.xctestBundleId];
+        if (!appInfo) {
+            log.error(`${this.xctestBundleId} not found!`);
+            return;
+        }
+        const signerIdentifier = appInfo.SignerIdentity;
+        log.debug('SignerIdentifier:', signerIdentifier);
+        const appContainer = appInfo.Container;
+        const execName = appInfo.CFBundleExecutable;
+        if (!execName.endsWith('-Runner')) {
+            log.error('Invalid CFBundleExecutable:', execName);
+            return;
+        }
+        const targetName = execName.substr(0, execName.indexOf('-Runner'));
+        const xctestPath = `/tmp/${targetName}-${sessionIdentifier.toUpperCase()}.xctestconfiguration`;
+        const xctestConfiguration = new XCTestConfiguration({
+            testBundleURL: `file://${appInfo.Path}/PlugIns/${targetName}.xctest`,
+            sessionIdentifier,
+            targetApplicationBundleID: this.targetBundleId,
+            productModuleName: this.productModuleName
+        });
+        const xctestContent = xctestConfiguration.getBytes();
+        const houseArrestService = await startHouseArrestService(this.udid);
+        const vendContainer = await houseArrestService.vendContainer(this.xctestBundleId);
+        const list = await vendContainer.listDirectory('/tmp');
+        for (let file of list) {
+            if (file.endsWith('.xctestconfiguration')) {
+                log.debug('removing /tmp/' + file);
+                await vendContainer.deleteDirectory('/tmp/' + file);
```

Comment:
could we remove all files at the same time? Example pseudocode:
```
const deletePromises = [];
for (const filename of filenames) {
  deletePromises.push(vendContainer.deleteDirectory(`/tmp/${filename}`))
}
if (!_.isEmpty(deletePromises)) {
  await B.all(deletePromises)
}
```

Reply:
There are not supposed to have more than one file if everything goes right.
Besides, I worry that multi-request from one afc services would cause connection stability issue